pilot/pkg/model/context.go

```diff
@@ -162,7 +162,7 @@ const (
 	DiscoveryPlainAddress = "istio-pilot:15007"
 
 	// IstioIngressGatewayName is the internal gateway name assigned to ingress
-	IstioIngressGatewayName = "istio-k8s-ingress-gateway"
+	IstioIngressGatewayName = "istio-ingress"
```

Comment:
this should be istio-k8s-ingress.. This is not meant to be referenced in user created virtual services. Also, ingress as a concept exists ONLY in k8s. Not in cloud foundry. It would be bad to force this logic on them as well.

Reply:
I would keep it consistent with the name of the ingress pod. Ingress is not really specific to k8s - as Istio proves, it can 
exists everywhere (at least the gateway equivalent of ingress). The word 'ingress' is generic.

I would expect non-k8s deployments to still use a Gateway named istio-ingress and provide equivalent functionality.

Reply:
Sure, but keep in mind that they do not have the need to merge our gateway with platform ingress. This name is for all intents and purposes a UUID that we use to create a unique name for the gateway. 
IF we run same istio across two platforms (one of which is kubernetes) and the other tries to use a gateway called "istio-ingress", this will cause failure, as kubernetes's default ingress controller will take over that name.